performance_regression_gradual_grow_throughput.py

```diff
@@ -0,0 +1,236 @@
+from enum import Enum
+from collections import defaultdict
+
+import json
+# from sdcm.send_email import Email
+from performance_regression_test import PerformanceRegressionTest
+from sdcm.results_analyze import ThroughputLatencyGradualGrowPayloadPerformanceAnalyzer
+
+
+class CSPopulateDistribution(Enum):
+    GAUSS = "gauss"
+    UNIFORM = "uniform"
+
+
+class PerformanceRegressionGradualGrowThroughutTest(PerformanceRegressionTest):  # pylint: disable=too-many-instance-attributes
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.NUM_THREADS = 500  # pylint: disable=invalid-name
+        self.CLUSTER_SIZE = self.params.get("n_db_nodes")  # pylint: disable=invalid-name
+        self.REPLICATION_FACTOR = 3  # pylint: disable=invalid-name
+        self.DATASET_SIZE = 3000 * 1024 * 1024 * 1024  # pylint: disable=invalid-name
+        self.ROW_SIZE_BYTES = 210 * 1024 * 1024 * 1024 / 720_000_000  # pylint: disable=invalid-name
+        self.ROW_COUNT = int(self.DATASET_SIZE / self.ROW_SIZE_BYTES /  # pylint: disable=invalid-name
+                             self.REPLICATION_FACTOR)
+        # How many standard deviations should span the cluster's memory
+        self.CONFIDENCE = 6  # pylint: disable=invalid-name
+        # intance i3.4xlarge
+        self.INSTANCE_MEMORY_GB = 122  # pylint: disable=invalid-name
+        self.GAUSS_CENTER = self.ROW_COUNT // 2  # pylint: disable=invalid-name
+        self.GAUSS_SIGMA = int(self.INSTANCE_MEMORY_GB * self.CLUSTER_SIZE * 1024 * 1024 *  # pylint: disable=invalid-name
+                               1024 // (self.CONFIDENCE * self.ROW_SIZE_BYTES * self.REPLICATION_FACTOR))
+
+        self.MAX_95TH_LATENCY = 50.0  # pylint: disable=invalid-name
+
+    def test_mixed_gradual_increase_load(self):  # pylint: disable=too-many-locals
```

Comment:
you are repeating the test 3 times... so i recommend you to have a "base class" that will do the things that are being repeated, and one for each workload, passing the right arguments (2, IIUC):

```suggestion
    def test_gradual_increase_load(self, stress_cmd, test_name):
        stress_num = 1
        num_loaders = len(self.loaders.nodes)

        self.run_fstrim_on_all_db_nodes()
        # run a write workload as a preparation
        compaction_strategy = self.params.get('compaction_strategy')
        self.preload_data(compaction_strategy=compaction_strategy)
        self.wait_no_compactions_running(n=400, sleep_time=120)
        self.run_fstrim_on_all_db_nodes()

        self.run_gradual_increase_load(stress_cmd_templ=stress_cmd,
                                       start_ops=self.start_ops,
                                       throttle_step=self.throttle_step,
                                       max_ops=self.max_ops,
                                       stress_num=stress_num,
                                       num_loaders=num_loaders,
                                       compaction_strategy=compaction_strategy,
                                       test_name=test_name)
```

and then the call for mixed workload will be:
```
def test_mixed_gradual_increase_load(self):
    test_gradual_increase_load(self, self.params.get('stress_cmd_m'), "Test 'mixed: read:50%,write:50%'")
```

Reply:
done